File: dispatch/python/module/scheduling/schedule_once.py

from module.scheduling.schedule_description import ScheduleDescription
from datetime import datetime

class ScheduleOnce(ScheduleDescription):
    def __init__(self, execution_time: datetime):
        self.execution_time = execution_time

    @classmethod
    def now(cls):
        return cls(datetime.now())

    def does_runnable_have_to_be_repeated(self):
        return False

    def update_execution_time(self):
        # Does not exist for ScheduleOnce.
        # A ScheduledRunnable with ScheduleDescription
        # of type ScheduleOnce is not supposed to be repeated.
        pass

    def get_execution_time(self):
        return self.execution_time

File: dispatch/ios/CLAID/Sources/CLAID/Module/Scheduling/ScheduleOnce.swift
//
//  ScheduleOnce.swift
//  CLAID
//
//  Created by Patrick Langer on 19.02.2025.
//


import Foundation
public actor ScheduleOnce: ScheduleDescription {
    private var executionTime: Date

    public init(executionTime: Date) {
        self.executionTime = executionTime
    }

    public static func now() -> ScheduleOnce {
        return ScheduleOnce(executionTime: Date())
    }

    public func doesRunnableHaveToBeRepeated() async -> Bool {
        return false
    }

    public func updateExecutionTime() async throws {
        // Does not exist for ScheduleOnce.
        // A ScheduledRunnable with ScheduleDescription
        // of type ScheduleOnce is not supposed to be repeated.
    }

    public func getExecutionTime()  async -> Date {
        return executionTime
    }
}
